src/Parser/Core/Modules/Items/InsigniaOfTheGrandArmy.js

```diff
@@ -0,0 +1,198 @@
+      this.damage += (event.amount || 0) + (event.absorbed || 0) + (event.overkill || 0);
+      this.chaoticDarknessDamage += (event.amount || 0) + (event.absorbed || 0) + (event.overkill || 0);
+    }
+    if (spellID === SPELLS.TORMENT_THE_WEAK_DAMAGE.id) {
+      this.damage += event.amount + (event.absorbed || 0);
+      this.tormentTheWeakDamage += event.amount + (event.absorbed || 0);
+    }
+    if (spellID === SPELLS.DARK_SORROWS_DAMAGE.id) {
+      this.damage += event.amount + (event.absorbed || 0);
+      this.darkSorrowsDamage += event.amount + (event.absorbed || 0);
+    }
+
+  }
+  on_byPlayer_absorbed(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.REFRACTIVE_SHELL_BUFF.id && spellId !== SPELLS.HOLY_BULWARK.id) {
+      return;
+    }
+    if (spellId === SPELLS.REFRACTIVE_SHELL_BUFF.id) {
+      this.healing += event.amount;
+      this.refractiveHealing += event.amount;
+    }
+    if (spellId === SPELLS.HOLY_BULWARK.id) {
+      this.healing += event.amount;
+      this.secureInTheLightHealing += event.amount;
+    }
+  }
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.INFUSION_OF_LIGHT_HEALING.id && spellId !== SPELLS.LIGHTS_EMBRACE_HEALING.id && spellId !== SPELLS.CHAOTIC_DARKNESS_HEALING.id && spellId !== SPELLS.SHADOWBIND_DAMAGE_HEALING.id) {
+      return;
+    }
+    if (spellId === SPELLS.INFUSION_OF_LIGHT_HEALING.id) {
+      this.healing += (event.amount || 0) + (event.absorbed || 0);
+      this.infusionOfLightHealing += (event.amount || 0) + (event.absorbed || 0);
+    }
+    if (spellId === SPELLS.LIGHTS_EMBRACE_HEALING.id) {
+      this.healing += (event.amount || 0) + (event.absorbed || 0);
+      this.lightsEmbraceHealing += (event.amount || 0) + (event.absorbed || 0);
+    }
+    if (spellId === SPELLS.SHADOWBIND_DAMAGE_HEALING.id) {
+      this.healing += (event.amount || 0) + (event.absorbed || 0);
+      this.shadowBindHealing += (event.amount || 0) + (event.absorbed || 0);
+    }
+    if (spellId === SPELLS.CHAOTIC_DARKNESS_HEALING.id) {
+      this.healing += (event.amount || 0) + (event.absorbed || 0);
+      this.chaoticDarknessHealing += (event.amount || 0) + (event.absorbed || 0);
+    }
+
+  }
+
+  get averageCritFromRing() {
+    return (((this.combatants.selected.getBuffUptime(SPELLS.SHOCKLIGHT_BUFF.id) / this.owner.fightDuration) * CRIT_AMOUNT * this.combatants.selected.traitsBySpellId[SPELLS.SHOCKLIGHT_TRAIT.id]) / 3).toFixed(2);
+  }
+
+  get averageVersFromRing() {
+    return (((this.combatants.selected.getBuffUptime(SPELLS.MURDEROUS_INTENT_BUFF.id) / this.owner.fightDuration) * VERSATILITY_AMOUNT * this.combatants.selected.traitsBySpellId[SPELLS.MURDEROUS_INTENT_TRAIT.id]) / 3).toFixed(2);
+  }
+
+  item() {
+    //Basics
+    let tooltip = this.damage + this.healing > 0 ? `This list will show how much of your total NLC trait contribution, the insignia was responsible for: <ul>` : ``;
```

Comment:
Nit: Maybe add a small bit about not including Light speed and Master of Shadows, even if for obvious reasons

Reply:
I'll actually just add baseline module and add them into this, even if they are just flat stat increases - getting that done tomorrow